db/python/tables/bq/function_bq_filter.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime
+from enum import Enum
+from typing import Any
+
+from google.cloud import bigquery
+
+from models.models import BillingColumn
+
+
+class FunctionBQFilter:
+    """
+    Function BigQuery filter where left site is a function call
+    In such case we need to parameterised values on both side of SQL
+    E.g.
+
+    SELECT ...
+    FROM ...
+    WHERE getLabelValue(labels, 'batch_id') = '1234'
+
+    In this case we have 2 string values which need to be parameterised
+    """
+
+    func_where = ''
+    func_sql_parameters: list[
+        bigquery.ScalarQueryParameter | bigquery.ArrayQueryParameter
+    ] = []
+
+    def __init__(self, name: str, implementation: str):
+        self.func_name = name
+        self.fun_implementation = implementation
+        # param_id is a counter for parameterised values
+        self._param_id = 0
+
+    def to_sql(
+        self,
+        column_name: BillingColumn,
+        func_params: str | list[Any] | dict[Any, Any],
+        func_operator: str = None,
+    ) -> tuple[str, list[bigquery.ScalarQueryParameter | bigquery.ArrayQueryParameter]]:
+        """
+        creates the left side of where :  FUN(column_name, @params)
+        each of func_params convert to BQ parameter
+        combined multiple calls with provided operator,
+        if func_operator is None then AND is assumed by default
+        """
+        values = []
+        conditionals = []
+
+        if not isinstance(func_params, dict):
+            # Ignore func_params which are not dictionary for the time being
+            return '', []
```

Comment:
Do you want an exception here?

Reply:
We potentially can throw exception, there is already check before calling this func to check if func_params is dict.